public/app/features/dashboard/components/PanelEditor/OptionsPaneCategory.tsx

```diff
@@ -112,6 +112,7 @@ export const OptionsPaneCategory = React.memo(
       <div
         className={boxStyles}
         data-testid="options-category"
+        data-plugin-sandbox={'panel-options'}
```

Comment:
This data-attribute won't affect at all plugins not running inside the sandbox. It'll be just another data-attribute.

Reply:
Will this wrap editors both added by the panel and by Grafana core with `data-plugin-sandbox`? Will that then allow plugins to tamper with any children of that dom element?

Would it work if we apply the `data-plugin-sandbox=${pluginId}` only for the elements added by the plugin instead? I think we could add a data attribute to those elements during panel configuration. 

So when the plugin developer calls `useFieldConfig` we could detect if any custom editors are added and then, under the hood, add a property that will be used while rendering that config editor.

Reply:
The plugin could potentially access elements in the panel options editor that were not directly render by itself but only and only panel editor options. Considering there's only one panel editor open at the time and it'll be the panel editor options of the plugin itself it sounds reasonable. Also it will definitely won't go any further up the dom than the panel options because it can't escape this attribute.

Having the pluginId as you suggest would be the best solution. I did want to put the $pluginId here instead of just "panel-options" but I the pluginId was not available in this component and it seems to be used in too many different places that also don't have the context of the plugin id and thus I went for a simpler solution using `panel-options` as the sandbox attribute value.

Reply:
Ah, ok ok! Let me have a quick look. I think we have access to the plugin in when calling the `useFieldConfig`. So I will see if we can define something like additional props during registration time. Will time box this and get back to you if my proposed solution will work.

Reply:
So I have two suggestions depending on which solution you prefer.

We add a "sandboxId" (or similar) to the OptionsEditorItem assign the pluginId to that id in createFieldConfigRegistry and then use it for the data attribute when rendering the editor in OptionsPaneItemDescriptor.
The other alternative is that we wrap the whole options pane section with the id. Then you can just use it from the PanelModel which is available when rendering the options pane.

Reply:
I followed your idea and ended up modifying the `OptionsPaneCategoryDescriptor` to accept a sandboxID. It'll be only passed and rendered with the plugin visualization options which are the ones provided by the plugin which are the ones we want to sandbox.